### ISSUE TITLE
Memory leak?

### ISSUE THREAD

**Issue by chris-martin** [tootsuite/mastodon#787]
Chrome reports the Mastodon webapp uses about 75 MB of memory when it starts, 200 MB after it's been running for a while, and - after I left it open overnight - 780 MB.

**Comment by wblondel** [comment]
I think that's because toots from the feed never get unloaded

**Comment by Gargron** [comment]
Correct. There is no user-land garbage collection on the Redux store so to speak. It's been bugging me for a while but I'm just running on the assumption that people occasionally close the page.

**Comment by Maverynthia** [comment]
I tend to leave stuff open sooo... not often I close a page.

**Comment by saper** [comment]
Wouldn't assume that either...

**Comment by lutoma** [comment]
I use mastodon in http://rambox.pro/, so it's pretty much open 24/7 for me.

**Comment by Geobert** [comment]
350Mo after 4hours and making the whole stuff very laggy. Typing a message is sluggish… not good user experience :-/

**Comment by clarfonthey** [comment]
Gonna bump this issue again since I get significant slowdowns if I don't refresh the app every once in a while. I think that the optimisation to cap the number of messages left in the feed after a while would be a good start, since I believe you can still reload more if you go down far enough? I end up with ~3k messages to load if I leave the app open overnight and then if I click that it lags so hard I can't easily type up a new post.

I think that capping at 1k messages would be a reasonable improvement, with the option to reload older posts as you read further.

This is probably a decently-sized architectural redesign for the frontend but it's definitely needed to avoid atrocious lag.

**Comment by phocks** [comment]
Have noticed it happening in Firefox too. Don't seem to have the same issue with Elk web interface.
